### ISSUE TITLE
Error on trying to load the database with KeyFile + Master Password

### ISSUE THREAD

**Issue by ShantanuShinde** [jorabin/KeePassJava2#38]
I have created a database using the KeePass 2 application, which takes a Master Password and a KeyFile. I want to open that database in Java, using this library. But I get the following error on the `load` method:

> Exception in thread "main" java.lang.IllegalStateException: Inconsistent stream start bytes. This usually means the credentials were wrong.
        at org.linguafranca.pwdb.kdbx.stream_3_1.KdbxSerializer.checkStartBytes(KdbxSerializer.java:129)
        at org.linguafranca.pwdb.kdbx.stream_3_1.KdbxSerializer.createUnencryptedInputStream(KdbxSerializer.java:88)
        at org.linguafranca.pwdb.kdbx.stream_3_1.KdbxStreamFormat.load(KdbxStreamFormat.java:37)
        at org.linguafranca.pwdb.kdbx.dom.DomDatabaseWrapper.<init>(DomDatabaseWrapper.java:57)
        at org.linguafranca.pwdb.kdbx.dom.DomDatabaseWrapper.load(DomDatabaseWrapper.java:62)
        at com.ni.apps.hardware.activation.test.DatabaseTest.<init>(DatabaseTest.java:18)

This is my code for trying to load the database:
```
InputStream databaseStream = getClass().getResourceAsStream("Database.kdbx");
InputStream keyStream = getClass().getResourceAsStream("Database.keyx");
KeyFile keyFile = new KeyFile("MyPassword".getBytes(), keyStream);
Database database = DomDatabaseWrapper.load(keyFile, databaseStream);
List<Entry> entries = database.findEntries("Sample Entry");
Entry entry = entries.get(0);
```
I have tried `SimpleDatabase` and `JaxbDatabase`, they give this 
error. I also tried a version 1.0 keyfile, gives the same error.

I am able to open the database with this password and file in the application fine.
Also, creating a database with just password works fine with the `KeePassJava2`. Only the Keyfile is the issue.
I have attached the test data with this issue
[Database.zip](https://github.com/jorabin/KeePassJava2/files/9497709/Database.zip)

**Comment by jorabin** [comment]
Hi @ShantanuShinde thanks for this report. 

I've finally got round to updating KeePassJava2 and to reviewing issues.

First of all KeyFile is only usable with KeePass V1 files which are binary. 


From KeePass V2 onwards Keyfiles are XML and you load a keyfile as follows:

            KdbxCreds creds = new KdbxCreds("MyPassword".getBytes(), keyStream);

However, that doesn't work either, in KeePassHava2 as the keyfile you are using is an XML KeyFile version 2 which I think was introduced around the time that Keepass file format was updated to Version 4.1. All these version numbers, but the long and short is that KdbxCreds needs to be updated to read KeyFile Version 2. Or at least check the version number.

Since it's pouring with rain here in London this afternoon, I will make that change and make a further comment when I have committed it.
